src/test/java/com/atharvadholakia/password_manager/controller/CredentialControllerTests.java

```diff
@@ -26,36 +26,46 @@ public class CredentialControllerTests {
 
   @MockBean private CredentialService credentialService;
 
-  private Credential credential;
+  private String createJsonInput(Object serviceName, Object username, Object password) {
+    return String.format(
+        "{\"serviceName\" : %s , \"username\" : %s , \"password\" : %s }",
+        convertToJsonValue(serviceName),
+        convertToJsonValue(username),
+        convertToJsonValue(password));
+  }
 
-  @BeforeEach
-  public void setup() {
-    credential = new Credential("TestserviceName", "Testusername", "Testpassword");
+  private String convertToJsonValue(Object value) {
+    if (value == null) {
+      return "null";
+    } else if (value instanceof String) {
+      return String.format("\"%s\"", value);
+    } else {
+      return value.toString();
```

Comment:
The `convertToJsonValue` method, as currently implemented, converts any non-null value to its string representation using `value.toString()`. This approach works fine for strings, but it has implications for other data types like `boolean` and `number` (e.g., int, double).

One potential Issue I see is, **Boolean and Number Handling:** 
When a boolean or number is passed to convertToJsonValue, it will be converted to a string (e.g., true becomes "true", 42 becomes "42"). In a JSON context, boolean values should be represented without quotes (true, false), and numbers should also be without quotes (42, 3.14). If they are converted to strings with quotes, they will be treated as string values by the JSON parser, which might lead to unexpected behavior or test failures.

Can you please check this point? If I'm missing something here, let me know.

Reply:
Yes, when a boolean or a number value is passed it will indeed be converted to a string. However, I have made a custom validator which validates this i.e a boolean string or a number only string will throw an exception. I had to do this because  JSON converts a boolean or number to string on its own anyways. Check file "StringTypeValidator.java" line 21 to 29.